### ISSUE TITLE
Failed calling webhook "validate.nginx.ingress.kubernetes.io" error while deploying ingress on RKE clusters with v1.21.1-rancher1-1

### ISSUE THREAD

**Issue by thehejik** [rancher/rancher#33067]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**
* TL;DR - Deploy custom RKE1 cluster with k8s `v1.21.1-rancher1-1` and try to create nginx-ingress based ingress - it will fail on:
`Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "validate.nginx.ingress.kubernetes.io": Post "https://ingress-nginx-controller-admission.ingress-nginx.svc:443/networking/v1beta1/ingresses?timeout=10s": dial tcp 10.43.163.232:443: connect: connection refused`

Or try to deploy rancher on RKE1 cluster with `v1.21.1-rancher1-1`:

* provision single VM with all the prerequisities for RKE1 clusters - I was trying ubuntu 18.04 in CI and sles15sp2 manually
* download latest (unstable) `rke-1.3.0-rc3` with support for k8s `v1.21.1-rancher1-1`
* prepare cluster.yaml with following content (~ borrowed from our CI):
``` yaml
kubernetes_version: v1.21.1-rancher1-1
nodes:
  - address: 192.168.122.61
    user: sles
    role: [etcd, controlplane,worker]
addon_job_timeout: 120
```
* call `rke-1.3.0-rc3 up --config cluster.yaml`
* create needed resources and install cert-manager and rancher:
``` bash
export KUBECONFIG=$PWD/kube_config_cluster.yml
kubectl create ns cattle-system
kubectl create ns cert-manager
helm install cert-manager jetstack/cert-manager --namespace cert-manager --version v1.3.1 --set installCRDs=true
sleep 60
helm upgrade --install=true rancher rancher-latest/rancher -n cattle-system --version=2.5.8 --set hostname=rancher.second --set replicas=1 --set rancherImageTag=master-head --set rancherImagePullPolicy=Always
```
**Result:**
helm will fail on attemp to install rancher on following error:
```
Release "rancher" does not exist. Installing it now.
Error: Internal error occurred: failed calling webhook "validate.nginx.ingress.kubernetes.io": Post "https://ingress-nginx-controller-admission.ingress-nginx.svc:443/networking/v1beta1/ingresses?timeout=10s": dial tcp 10.43.241.218:443: connect: connection refused
```

**Other details that may be helpful:**
* Custom RKE1 clusters with k8s `v1.21.1-rancher1-1` are also affected, simple reproducer:
```
kubectl apply -f - <<EOF
 apiVersion: networking.k8s.io/v1beta1
 kind: Ingress
 metadata:
   name: ingress-myservicea
   annotations:
     kubernetes.io/ingress.class: "nginx"
 spec:
   rules:
   - host: rancher.second
     http:
       paths:
       - path: /
         backend:
           serviceName: myservicea
           servicePort: 80
 EOF
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "validate.nginx.ingress.kubernetes.io": Post "https://ingress-nginx-controller-admission.ingress-nginx.svc:443/networking/v1beta1/ingresses?timeout=10s": dial tcp 10.43.163.232:443: connect: connection refused
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): `rancher/rancher:master-head b2f8824` (at the time of the attempt)
- Installation option (single install/HA): single or HA / any

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): local
- Machine type (cloud/VM/metal) and specifications (CPU/memory): any
- Kubernetes version (use `kubectl version`): v1.21.1-rancher1-1

**Comment by thehejik** [comment]
Found upstream ingress-nginx issue https://github.com/kubernetes/ingress-nginx/issues/5401

Quick workaround is to run `kubectl delete -A ValidatingWebhookConfiguration ingress-nginx-admission` and then rancher can be installed.

**Comment by thehejik** [comment]
Custom RKE clusters with k8s `v1.21.1-rancher1-1` are also affected.

**Comment by kinarashah** [comment]
Available to test with RKE `v1.3.0-rc5` and Rancher master-head once build passes https://drone-publish.rancher.io/rancher/rancher/5699

**Comment by thehejik** [comment]
Now the `rancher/rancher:master-head 62a8e6a` can be now deployed over helm on RKE1 cluster with k8s `v1.21.1-rancher2-1` - the cluster was provisioned by using `rke-1.3.0-rc6` binary and even user ingresses defined in default namespace are working fine.

Problem is with custom RKE1 (provisioned from rancher) and with imported RKE1 cluster deployed by `rke-1.3.0-rc6` (both with `v1.21.1-rancher2-1`). The ingresses can be created without any error but the forwarding to a workload doesn't work for some reason.

@kinarashah Do you have any idea why the ingress doesn't work on custom/imported RKE1 clusters? 
I noticed one suspicious thing that dns records for the ingresses nor for any workload are not created automatically anymore in custom clusters, could be this the reason?

**Comment by thehejik** [comment]
Blocked by https://github.com/rancher/rancher/issues/32801

**Comment by kinarashah** [comment]
Available to test once build passes https://drone-publish.rancher.io/rancher/rancher/5719

**Comment by thehejik** [comment]
**Test report:**
Tested on `rancher/rancher:master-head 5715b8a` with custom RKE1 cluster running k8s `v1.21.1-rancher2-1`

* Ingresses can be created for custom and local clusters
* ingresses are working when custom cluster has `Nginx Ingress` and `Nginx Default Backend` flags enabled (via `Edit` Cluster button / `Advanced Options`)
* custom rewrite rules in ingress Annotations like `nginx.ingress.kubernetes.io/rewrite-target: /` are working (eg. for target Path `/test`)
* One problem found that corresponding generated ingress services like `ingress-53aca48d52db91d9b565d5cbf97e80a0` are not getting removed after deleting the ingress https://github.com/rancher/rancher/issues/33178

Close.

**Comment by jayjaytay** [comment]
Also seeing this with RKE2 1.21.3.  Had to revert to 1.20.7 to avoid the timeout.

**Comment by marty90** [comment]
+1, with RKE rke version v1.3.1 and v1.21.5-rancher1-1

**Comment by dmpe** [comment]
Can also confirm, this is not fixed yet.

**Comment by penumbra23** [comment]
Can confirm on RKE2 `v1.22.6+rke2r1` the issue still persists.

**Comment by tacerus** [comment]
I encounter this as well - weirdly enough it only started happening after it had been working fine for some days after installing the cluster.

**Comment by mveroone** [comment]
Hi.
Here, killing my canal pods to have them recreated fixed the issue.
I haven't dug deeper, but the cause was that the server node podes couldn't each pods on another node
